src/main/java/io/gatling/demostore/website/controllers/CartController.java

```diff
@@ -1,61 +1,89 @@
 package io.gatling.demostore.website.controllers;
 
+import javax.servlet.http.Cookie;
 import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpSession;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
 import java.util.HashMap;
+import java.util.Map;
 
-import io.gatling.demostore.models.Cart;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.gatling.demostore.models.CartEntry;
 import io.gatling.demostore.models.ProductRepository;
 import io.gatling.demostore.models.data.Product;
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.*;
 
 @Controller
 @RequestMapping("/cart")
 @SuppressWarnings("unchecked")
 public class CartController {
 
+    public static final class Carts {
```

Comment:
nit: Should not be in Controller.

`public static final class` => should be a top class in its own file